mars/tensor/special/core.py

```diff
@@ -112,3 +115,54 @@ def execute(cls, ctx, op):
                 if ret.dtype != op.dtype:
                     ret = ret.astype(op.dtype)
                 ctx[op.outputs[0].key] = ret
+
+
+class TensorTupleOp(TensorSpecialUnaryOp):
+    @property
+    def output_limit(self):
+        return self._func_outputs
+
+    def __call__(self, x, out=None):
+        x = astensor(x)
+
+        if out is not None and not isinstance(out, ExecutableTuple):
+            raise TypeError(
+                f"out should be ExecutableTuple object, got {type(out)} instead"
+            )
+
+        func = getattr(spspecial, self._func_name)
+        res = func(np.ones(x.shape, dtype=x.dtype))
+        res_tensors = self.new_tensors(
+            [x],
+            kws=[
+                {
+                    "side": f"{self._func_name}[{i}]",
+                    "dtype": res[i].dtype,
+                    "shape": res[i].shape,
+                }
+                for i in range(len(res))
+            ],
+        )
+
+        if out is None:
+            return ExecutableTuple(res_tensors)
+
+        for i in range(len(res_tensors)):
+            out[i].data = res_tensors[i].data
+        return out
+
+    @classmethod
+    def execute(cls, ctx, op):
+        inputs, device_id, xp = as_same_device(
+            [ctx[c.key] for c in op.inputs], device=op.device, ret_extra=True
+        )
+
+        with device(device_id):
+            with np.errstate(**op.err):
+                if op.is_gpu():
+                    ret = cls._execute_gpu(op, xp, inputs[0])
+                else:
+                    ret = cls._execute_cpu(op, xp, inputs[0])
+
+                for i in range(len(op.outputs)):
+                    ctx[op.outputs[i].key] = ret[i]
```

Comment:
Loop over `op.outputs` itself, not indices.

```suggestion
                for output in op.outputs:
                    ctx[output] = ret[i]
```